### ISSUE TITLE
finish and merge support for Exchange 2013

### ISSUE THREAD

**Issue by pabs3** [openchange/openchange#314]
There is semi-working support for Exchange 2013 in the following branches. It would be great if it could be finished and merged.

https://github.com/openchange/openchange/
jkerihuel/ncacn_http-samba-4.2

https://github.com/jkerihuel/openchange/
jkerihuel/ncacn_http

https://github.com/kernevil/openchange/
ncacn_http

Steps needed to solve this:

Setup an Exchange 2013 server for testing the feature.

Rebase/merge the branches onto master.

Resolve the samba-debug library issue as it prevents building openchange. This issue may be resolved already, not sure.

Fix the libmapi ABI breakage due to inserting a member into the middle of the mapi_profile struct. Only adding new members at the end of the struct is the way to fix this.

Make the openchangeclient --username parameter work again.

Fix the mapitest OXCSTOR-LOGON OpenMsgStore Unknown Error (0x0000000b) and the errors for 87 other tests, probably not all of them are relevant though.

**Comment by KARTHIK5989** [comment]
Can you please let me know whether which version of open change tool supports Exchange 2k13 ?

And please provide  documents on  how to install openchange tool Over Exchange 2k13 ?

Requirements like Linux OS ? Version ?

**Comment by bradh** [comment]
Not supported. I'm no longer working on this.

**Comment by chris001** [comment]
What's the roadmap or plan to merge  this and fix?  Just asking... because it seems like it was ready 8 months ago, for testing and just needs to start the merging...

**Comment by bradh** [comment]
I do not know of anyone else working on this, and as such, there is no roadmap or plan beyond eventually fixing the logging stuff. No schedule that I'm aware of.

**Comment by KARTHIK5989** [comment]
Can you please let me know whom I need to contact with his mail id regarding open change tool support for Exchange 2k13 ? We required it in our test environment

**Comment by chris001** [comment]
Hi @KARTHIK5989  I'd imagine there is a chance that @pabs3 could do the merge which he outlined in above.  This would be really nice!  I'm tagging him so he'll hopefully join in this conversation

**Comment by KARTHIK5989** [comment]
Hi @pabs3 , May I know whether there is any plan to support Ex2k13 ? Please let us know we are looking for it .

**Comment by pabs3** [comment]
I'm not part of the OpenChange project so don't know what their plans are for supporting Exchange 2013 but I don't think anyone has the time, interest and skills to fix up the work that Brad did and get it merged.

The OpenChange project seems to be a bit of a dead end since it can't even be compiled with the latest Samba versions and has thus been removed from the development versions of the Debian and Ubuntu distributions.

If I were you I would look at getting rid of Exchange instead of supporting it longer.

**Comment by KARTHIK5989** [comment]
Any idea whether openchange works for Mapi-Rpc-https connections . I installed and trying to work with 

Openchange release (2.3 "Vulcan") but I didn't find any RPC- http options available.

**Comment by chris001** [comment]
Yes.  MAPI RPC over HTTP (HTTPS) is the old name, new name is "Outlook Anywhere".
Supported 3 years ago, January 2013, in Openchange server 2.0
http://openchange.org/developers/relnotes/2.0-quadrant.html

Here is where the support was added... It mentions the section of the config file which you must add..
https://github.com/openchange/openchange/pull/40
"Added option to configure Outlook Anywhere with SSL and set an external hostname.
    It requires new section: [autodiscover:rpcproxy]."

Note MAPI-RPC-HTTPS (Outlook Anywhere) is different from MAPI-HTTP(S) (Outlook 2016) see Julien's comment for more detail..
https://github.com/openchange/openchange/issues/412

**Comment by KARTHIK5989** [comment]
Actually I am looking for Openchange client which connects to Microsoft Exchange 2k10 using Rpc-https .
 Hence followed the blow link and able to install successfully with openchange client  version ( 2.3-GIT-36a22ee (Vulcan) )   in Redhat 6.5 . But I couldn't able to find any options to set as Rpc-https client  
http://openchange.org/documentation/howto/openchangeclient_setup.html 

Could you please help me  how to setup this to work as Rpc-https client using openchange client . Any document will be really helpful

**Comment by bradh** [comment]
It isn't merged, and it isn't likely to be in its current state. 

There is my consolidated code at https://github.com/bradh/openchange/tree/RoH but I'm not supporting it, or helping people install it - I don't have time.

I mean it.

I think there is possibly a bug in the underlying samba rpc-http code that causes wedged connections at least in in test scenarios.

**Comment by chris001** [comment]
Here's some information.  It's not entirely clear.  Some text says it's working.  Others say not yet.
https://sambaxp.org/archive_data/SambaXP2014-DATA/thu/track1/Julien_Kerihuel-OpenChange_BeyondTechnicalFulfillment.pdf

Note: mapi rpc https seems to be related to rpcproxy and nacn_http, mentioned by julien (founder of openchange) as "not yet merged".

http://sogo.nu/files/docs/SOGo%20Native%20Microsoft%20Outlook%20Configuration.pdf
Page 9-16 show configuration.

Maybe just installing Zentyal Community would be easier and just take 3 minutes instead of 3 hours of manually configuring Samba 4 / openchange / dovecot / etc! Depending on what you're trying to do.

**Comment by chris001** [comment]
Inverse is offering in the sogo documentation to fix these issues on a consulting basis if you need them fixed sooner than waiting for them to be fixed.  They would probably accept your payment and then engage with Zentyal / Openchage developers, to fix whichever issues you need working.
"The current version of the Native Microsoft Outlook compatibility layer has some limitations.
Those limitations will be overcome in the upcoming releases. If you are interested in having those
limitations fixed more rapidly, please contact Inverse by sending an email to support@inverse.ca "
